sdk/dapp-kit/src/components/styling/StyleMarker.tsx

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Slot } from '@radix-ui/react-slot';
+import type { ComponentPropsWithoutRef, ElementRef, ReactNode } from 'react';
+import { forwardRef } from 'react';
+
+import { styleDataAttribute } from '../../constants/styleDataAttribute.js';
+
+import './StyleMarker.css.js';
+
+type StyleMarker = {
+	children: ReactNode;
+};
+
+export const StyleMarker = forwardRef<
```

Comment:
I called this `StyleMarker` because the same data attribute will also be used for scoping theme variables which are used for styling